core/shared/src/main/scala/zio/logging/LogGroup.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2019-2022 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio.logging
+
+import zio.{ FiberRefs, LogLevel, Trace, Zippable }
+
+trait LogGroup[A] { self =>
```

Comment:
This is where you could explore adding:

```suggestion
trait LogGroup[A] { self =>
  def relation: LogGroupEquivalence[A]
```

Thus, every log group would both provide a group mapping to some type `A`, as well as a relation on values of type `A`. This sort of makes sense because in order to use the `A`, we need some capability on it (the relation), so bundling it directly inside is similar to how `FiberRef` avoids type classes by bundling fork/join operations inside it.

This slightly complicates `++`, which must take an `Unzippable`, so it can extract out the pieces and provide them to the relation.